pkg/server/updateResults.go

```diff
@@ -0,0 +1,87 @@
+package server
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/stolostron/recommends/pkg/kruize"
+	"github.com/stolostron/recommends/pkg/model"
+	klog "k8s.io/klog/v2"
+)
+
+var UpdateQueue chan CreateExperiment
+
+type UpdateResults struct {
+	Version           string                          `json:"version"`
+	ExperimentName    string                          `json:"experiment_name"`
+	StartTimestamp    string                          `json:"interval_start_time"`
+	EndTimestamp      string                          `json:"interval_end_time"`
+	KubernetesObjects []model.KubernetesObjectMetrics `json:"kubernetes_objects"`
+}
+
+func init() {
+	UpdateQueue = make(chan CreateExperiment)
+}
+
+//var update_results_url = config.Cfg.KruizeURL + "/updateResults"
+
+func ProcessUpdateQueue(q chan CreateExperiment) {
+	for {
+		klog.Info("Processing update Q")
+		ce := <-q
+		updateResultRequest(&ce)
+		klog.Infof("Processed %s", ce.ExperimentName)
+	}
```

Comment:
This loop is serializing the requests.
Use `go updateResultRequest(&ce)` to parallelize the processing.